drivers/crc/crc_hw.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CRC_H
+#define ZEPHYR_INCLUDE_DRIVERS_CRC_H
+
+#include "autoconf.h"
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct crc_data
+ * @brief Runtime driver configuration
+ * @param mutex Zephyr mutex object to make driver thread-safe when driver
+ *				is used within multiple threads
+ */
+struct crc_data {
+#ifdef CONFIG_CRC_THREADSAFE
+	struct k_mutex mutex;
+	k_timeout_t crc_8_timeout;
+	k_timeout_t crc_32_timeout;
+#endif /* CONFIG_CRC_THREADSAFE */
+};
+
+#ifdef CONFIG_CRC_HW
```

Comment:
I have not seen such conditional exclusion in other driver API headers, and it appears to go against [the API Guidelines](https://docs.zephyrproject.org/latest/develop/api/design_guidelines.html#conditional-data-and-apis).